### PR TITLE
Add experimental settings type

### PR DESCRIPTION
**Pull request by rzhao271**

### CODE REVIEW

**Comment by rzhao271** [comment]
Looks like I accidentally checked in a change to toggle.ts. Editing.

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 1]
```diff
@@ -659,6 +668,7 @@ const SETTINGS_BOOL_OBJECT_TEMPLATE_ID = 'settings.boolObject.template';
 const SETTINGS_COMPLEX_TEMPLATE_ID = 'settings.complex.template';
 const SETTINGS_NEW_EXTENSIONS_TEMPLATE_ID = 'settings.newExtensions.template';
 const SETTINGS_ELEMENT_TEMPLATE_ID = 'settings.group.template';
+const SETTINGS_EXTENSION_TOGGLE_TEMPLATE_ID = 'settings.boolWithEnable.template';
```
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:671]: TODO: Change ID name

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 2]
```diff
@@ -1878,6 +1892,112 @@ export class SettingBoolRenderer extends AbstractSettingRenderer implements ITre
 	}
 }
 
+type ExtensionToggleSettingTelemetryClassification = {
+	action: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'The type of action performed by the setting button' };
+	owner: 'rzhao271';
+	comment: 'Event used to gain insights into when users are using the experimental service enable setting';
```
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:1898]: TODO: change comment

**Review by sandy081 — CHANGES_REQUESTED** [review]
General design feedback: This feature is nothing to do with the existing configurations. If I understand correctly they are like feature flags and toggling them will perform a different action based on how the feature is configured. For eg: Installing an extension. Please let me know if there are other action types. My suggestion is not to marry this with the configuration model  and make it only Settings UI concept. By that I mean, read these experimental feature flags directly in the settings editor and render them accordingly. It is up to you where to read them in the settings editor land, for eg., you can  make these experimental flags as part of the view model (PreferencesModel). This way the real configuration model is untouched and won't have these fake settings. 

May I know if the action is to install an extension, what is the user flow?

**Comment by rzhao271** [comment]
The UI contains a single button that the user can press. If the extension isn't installed, the button installs the extension. If the extension isn't enabled, the button enables the extension. If the extension is enabled, the button disables the extension. The text of the button changes depending on the state of the extension.

**Comment by sandy081** [comment]
Can you please show how such buttons look like? I think the button has to say Install **A** extension from **B** so that user is aware which extension and who has published it. I do not think you want to build another extensions view in settings editor by providing install / reload / enable actions there. Have you thought about simply redirecting to extensions view and showing the extension?

**Review by roblourens — DISMISSED** [review]
I think it still bugs me that this thing looks like a setting and has a fake setting ID. I would rather see some real title that isn't trying to look like a setting, that makes it a little confusing about what exactly that row means. But this is ok

**Pull request closed without merging**